Source/Extensions/Blazorise.TreeView/Internal/_TreeViewNode.razor.cs

```diff
@@ -0,0 +1,207 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Blazorise.Extensions;
+using Blazorise.TreeView.Extensions;
+using Blazorise.Utilities;
+using Microsoft.AspNetCore.Components;
+#endregion
+
+namespace Blazorise.TreeView.Internal;
+
+public partial class _TreeViewNode<TNode> : BaseComponent
+{
+    #region Methods
+
+    protected override void BuildClasses( ClassBuilder builder )
+    {
+        builder.Append( "b-tree-view-node" );
+        builder.Append( "b-tree-view-node-collapsed", !Expanded );
+
+        base.BuildClasses( builder );
+    }
+
+    private async Task ToggleNode( TreeViewNodeState<TNode> nodeState, bool refresh = true )
+    {
+        nodeState.Expanded = !nodeState.Expanded;
+
+        if ( nodeState.Expanded )
+        {
+            if ( !ExpandedNodes.Contains( nodeState.Node ) )
+            {
+                ExpandedNodes.Add( nodeState.Node );
+            }
+
+            await ExpandedNodesChanged.InvokeAsync( ExpandedNodes );
+
+            if ( nodeState.HasChildren )
+            {
+                var childNodes = GetChildNodesAsync is not null
+                    ? await GetChildNodesAsync( nodeState.Node )
+                    : GetChildNodes( nodeState.Node );
+
+                if ( !nodeState.Children.Select( x => x.Node ).AreEqual( childNodes ) )
+                {
+                    nodeState.Children = new();
+
+                    await foreach ( var childNodeState in childNodes.ToNodeStates( HasChildNodesAsync, HasChildNodes, ExpandedNodes.Intersect( childNodes ?? Enumerable.Empty<TNode>() ).Any() ) )
```

Comment:
It seems that your link indicates `Intersect` is faster than `Any`. Or am I wrong?

Reply:
Reason why I searched and linked is because I believe any is more intuitive and readable here. As for performance considerations it is said that it depends on the data, plus intersect has an additional memory footprint. We could possibly benchmark but I guess it's no big problem to maintain one or the other.